### PR TITLE
log: Add syslog support

### LINKED ISSUE #22: log: Enable syslog logging
It's important for debug that we centralize all logs to the same place. The easiest way for that is to use `syslog`. We will be able to check the logs by running the following command:
```bash
journalctl -f
```

### PR DESCRIPTION
**Pull request by sboeuf**
This PR first re-vendors the proxy, then it constraints the logrus version to be v1.0.4 instead of v1.0.3.
At last, the most important part of this PR is that we enable the proxy logging towards the syslog to centralize all proxy logs to the system journal.

Fixes #22 #24

### CODE REVIEW

**Comment by bergwolf** [comment]
LGTM

**Comment by sameo** [comment]
LGTM

File: proxy.go [hunk 1]
```diff
@@ -108,15 +107,27 @@ func unixAddr(uri string) (string, error) {
...
 func setupLogger(logLevel string) error {
 	level, err := logrus.ParseLevel(logLevel)
 	if err != nil {
 		return err
 	}
 
-	logrus.SetLevel(level)
+	proxyLog.SetLevel(level)
+
+	hook, err := lSyslog.NewSyslogHook("", "", syslog.LOG_INFO, "")
```
Comment (jodh-intel) [inline proxy.go]: The last arg here needs to be `proxyName` - see https://github.com/clearcontainers/proxy/pull/181.

**Comment by jodh-intel** [comment]
Dang - too late ;)

**Pull request merged**